core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -354,21 +354,26 @@ private void applyConfiguration(CreateContainerCmd createCommand) {
 
         Set<Link> allLinks = new HashSet<>();
         Set<String> allLinkedContainerNetworks = new HashSet<>();
+
+        List<com.github.dockerjava.api.model.Container> containers = dockerClient.listContainersCmd().exec();
+
+        List<InspectContainerResponse> inspectContainerResponses = containers.stream()
+                .map(container -> dockerClient.inspectContainerCmd(container.getId()).exec())
```

Comment:
this is a N+1 issue and might be **very** slow in environments with many running containers (like CI servers)

I would rather use `getContainerId` from `linkableContainer`